src/build_sysimg.jl

```diff
@@ -5,9 +5,19 @@
 # next to libjulia (except on Windows, where it goes in $JULIA_HOME\..\lib\julia)
 # Allow insertion of a userimg via userimg_path.  If sysimg_path.dlext is currently loaded into memory,
 # don't continue unless force is set to true.  Allow targeting of a CPU architecture via cpu_target
-@unix_only const default_sysimg_path = joinpath(dirname(Libdl.dlpath("libjulia")),"sys")
-@windows_only const default_sysimg_path = joinpath(JULIA_HOME,"..","lib","julia","sys")
-function build_sysimg(sysimg_path=default_sysimg_path, cpu_target="native", userimg_path=nothing; force=false)
+@unix_only function default_sysimg_path(debug=false)
+    joinpath(dirname(Libdl.dlpath(debug ? "libjulia-debug" : "libjulia")),
+             "julia", debug ? "sys-debug" : "sys")
+end
+
+@windows_only function default_sysimg_path(debug=false)
+    joinpath(JULIA_HOME, "..", "lib", "julia", debug ? "sys-debug" : "sys")
+end
+
+function build_sysimg(sysimg_path=nothing, cpu_target="native", userimg_path=nothing; force=false, debug=false)
```

Comment:
This could be written as `function build_sysimg(sysimg_path=default_sysimg_path(debug), cpu_target="native", userimg_path=nothing; force=false, debug=false)` and then remove the three lines below.